### ISSUE TITLE
How to inject field "display" names in validation error messages

### ISSUE THREAD

**Issue by jure123** [rxweb/rxweb#518]
We need help for @rxweb/reactive-form-validators

We require that all validation error messages display field friendly names in the text. For example:

- Field 'First name' is required
- Field 'First name' must contain at least 3 characters
- Field 'Company email' is not a valid email address

I know that we can define custom error messages for every validator, but for large forms with many validators this is a lot of work. If possible we would like to use generic error message strings (like "Field {{0}} is required"), where field's "friendly name" would be automatically injected in the error message.

This is an IMAGINARY (NON-WORKING) solution that may solve our problem:

First we could define global error messages with container for field friendly name:

```
ReactiveFormConfig.set({
  "validationMessage": {
    "required":"Field {{0}} is required.",
    "minLength": "Field {{0}} must contain at least {{1}} characters.",
    "email": "Field {{0}} is not a valid email address."
  }
});
```

Then in Decorator driven code we could define "display name" in @prop decorator, like so:

```
@prop({ displayName: "First name" })
@required()
@minLength({value: 3})
public firstname: string

@prop({ displayName: "Company email" })
@email()
public email: string
```

Then the error messages could be automatically generated with display names injected in the strings.
Also formGroup.getErrorSummary(true) could return all messages with display names injected:

```
- Field 'First name' is required
- Field 'First name' must contain at least 3 characters
- Field 'Company email' is not a valid email address

```
This feature would be very useful especially in formGroup.getErrorSummary() method because when showing all error messages on the page it's really necessary to show field friendly names.

Is there any other elegant way we can achieve the requested functionality on the current version of @rxweb/reactive-form-validators?

Thank you in advance for your consideration.

**Comment by ajayojha** [comment]
@jure123 Very good suggestion and will do little bit amendments like instead ```{{0}}``` we will use ```{{displayName}}``` for string interpolation because ```{{0}}``` is already reserved for current FormControl value.

Would be happy to add this feature in the next release :).

**Comment by jure123** [comment]
@ajayojha, I noticed that this project development has not been very active for the past 6 months. Are you planning to update it more actively in the future or you don't have plans for it? It would be helpful to know. Thanks for info.

**Comment by ajayojha** [comment]
Yes, this is active, will start working from next week… you will see the progress I. Next couple of weeks 😊

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by ajayojha** [comment]
@jure123 The feature has been released in the [2.1.7 version](https://www.npmjs.com/package/@rxweb/reactive-form-validators/v/2.1.7).

I have added a `messageNexus` object property instead of ```displayName```, because there are fair chances where we need to convert multiple properties text. I have added `messageNexus` property on the validation decorator level as well.

Here are the few examples:

Configuration on `prop` decorator level, but messageNexus object will be transpiled on every validation decorator failed message.
```js
@prop({ messageNexus:{displayName: "First Name"}  })
@minLength({value: 3})
public firstname: string
```
Configuration on specific validation decorator level.

```js
@required({messageNexus:{requiredFieldLabel:'First Name'}})
@minLength({value: 3})
public firstname: string
```

Configuration on specific validation decorator level as well as on property level, both object to be merge during message formation.
```js
@prop({ messageNexus:{displayName: "First Name"}  })
@required({messageNexus:{requiredFieldLabel:'First Name'}})
@minLength({value: 3})
public firstname: string
```
